fix: similar paths are broken

jncarter123 opened the pull request:
There was an issue with startsWith because it cannot tell the difference between similar paths.

For example electrical.switches.bank.25.1 and electrical.switches.bank.25.10 look the same to startsWith.

The new behaviour is to split the path and compare to the mapping.

jncarter123 committed:
fix: similar paths are broken

There was an issue with startsWith because it cannot tell the difference
between similar paths.

For example electrical.switches.bank.25.1 and electrical.switches.bank.25.10
look the same to startsWith.

The new behaviour is to split the path and compare to the mapping.

sbender9 commented:
I'd prefer to do this differently. I'd don't like all the slicing, joining, etc. for every path that comes in.

I would change to ensure at plugin start that all the mappings end with a dot. That way startsWith will still work and is much more efficient.

I can do that if you don't want to.

jncarter123 commented:
That sounds like a good solution. I can change it this evening.

jncarter123 committed:
Modified to append a dot(.) to both the delta path and mappings path.

This forces startsWith to look at the full path and works for both
1:1 mappings and 1:Many mappings.

1:1
water.2.pressure
watermaker.boost.pressure

1:1
generic.temperatures.userDefined129.0.temperature
environment.inside.waterheater.temperature

1:Many
electrical.switches.bank.21.3
electrical.switches.bank.dclc21.generatorStart
electrical.switches.bank.dclc21.generatorStart.state
electrical.switches.bank.dclc21.generatorStart.order
electrical.switches.bank.dclc21.generatorStart.breakerCurrent
